### DISCUSSION TITLE
Make a sphere particle system using raylib

### DISCUSSION THREAD

**Discussion by wlxmhls** [raysan5/raylib#2754] [Q&A] [answered]
Hello,

I want to make a 3D sphere particle system using raylib. The effect should like this(ignore the background color), and the particle should move dynamically, but forming a sphere shape.
![图片](https://user-images.githubusercontent.com/5212346/195531478-1862acea-7ee5-4111-bc4e-3178b8659d4d.png)

I have no idea how to make it ... Any ideas?

**Comment by chriscamacho** [comment, marked as answer]
how should they move? using what algorithm?

you could use a shader to do this, or simpler depth sorted billboards, using simple trig, have them at a set distance from the origin, and rotate on two axes, you can then mess with the two axis angles to have them move round the globe shape

here's an old blog post I did which will help you with billboards....

https://bedroomcoders.co.uk/raylib-billboards-advanced-use/

**Reply by wlxmhls** [reply]
Thanks! The billboard should work. I will try it.

**Comment by chriscamacho** [comment]
don't forget to post back here so we can all enjoy the effect !
I was mighty tempted to spin one up myself, but busy with a project...

**Reply by wlxmhls** [reply]
Sorry for so long delay. The effect is a particle picture rolling on the billboard. Maybe I will post it later.
